infrastructure/transform/pipeline.tf

```diff
@@ -0,0 +1,60 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# Get all directories that have pipeline.json in them and say that they are pipeline directories
+
+resource "azurerm_data_factory_pipeline" "pipeline" {
+  for_each        = { for pipeline in local.pipelines : pipeline.pipeline_dir => [pipeline.pipeline_json, pipeline.pipeline_parameters] }
```

Comment:
a little confused by why the json and parameters are different

Reply:
Refactored it out